test/Conversion/FIRRTLToHW/lower-to-hw.mlir

```diff
@@ -727,9 +727,11 @@ firrtl.circuit "Simple"   attributes {annotations = [{class =
     // CHECK-NEXT: %0 = sv.read_inout %reg : !hw.inout<i32>
     // CHECK-NEXT: %reg2 = sv.reg : !hw.inout<i32>
     // CHECK-NEXT: %1 = sv.read_inout %reg2 : !hw.inout<i32>
-    // CHECK-NEXT: sv.alwaysff(posedge %clock) {
-    // CHECK-NEXT: }(syncreset : posedge %reset) {
-    // CHECK-NEXT:    sv.passign %reg2, %c0_i32 : i32
+    // CHECK-NEXT: sv.always posedge %clock  {
+    // CHECK-NEXT:   sv.if %reset  {
+    // CHECK-NEXT:     sv.passign %reg2, %c0_i32 : i32
+    // CHECK-NEXT:   } else  {
+    // CHECK-NEXT:   }
```

Comment:
It would be better if the else block was only constructed if needed.

Reply:
I'm currently creating IfOp with `elseRegion` because `elseRegion` might be used in the feature. 
This is MLIR problem but I'm not sure how to create a region on demand. What is a conventional way to create a region on demand?

Reply:
I think this is fine to leave it as is, then. 👍